spring-kafka/src/main/java/org/springframework/kafka/listener/KafkaMessageListenerContainer.java

```diff
@@ -427,7 +429,21 @@ public void onPartitionsAssigned(Collection<TopicPartition> partitions) {
 						if (ListenerConsumer.this.logger.isDebugEnabled()) {
 							ListenerConsumer.this.logger.debug("Committing on assignment: " + offsets);
 						}
-						if (KafkaMessageListenerContainer.this.getContainerProperties().isSyncCommits()) {
+						if (ListenerConsumer.this.transactionTemplate != null) {
+							ListenerConsumer.this.transactionTemplate.execute(new TransactionCallbackWithoutResult() {
+
+								@SuppressWarnings({ "unchecked", "rawtypes" })
+								@Override
+								protected void doInTransactionWithoutResult(TransactionStatus status) {
+									((KafkaResourceHolder) TransactionSynchronizationManager
+											.getResource(ListenerConsumer.this.kafkaTxManager.getProducerFactory()))
```

Comment:
The `ListenerConsumer.this.transactionTemplate` might not be null, but `kafkaTxManager` can be. Because we might use some different impl.
See #373

I think we should add `&& ListenerConsumer.this.kafkaTxManager != null` to the `if`.

Let me know and I'll fix on merge!